api/src/main/java/org/ehcache/expiry/Expirations.java

```diff
@@ -108,36 +126,83 @@ public int hashCode() {
     @Override
     public String toString() {
       return this.getClass().getSimpleName() + "{" +
-          "create=" + create +
-          ", access=" + access +
-          ", update=" + update +
-          '}';
-    }
-
-    @Override
-    public Duration getExpiryForUpdate(Object key, ValueSupplier<?> oldValue, Object newValue) {
-      return update;
+             "create=" + create +
+             ", access=" + access +
+             ", update=" + update +
+             '}';
     }
   }
 
-  private static class TimeToLiveExpiry extends BaseExpiry {
+  private static class TimeToLiveExpiry extends BaseExpiry<Object, Object> {
     TimeToLiveExpiry(Duration ttl) {
       super(ttl, null, ttl);
     }
   }
 
-  private static class TimeToIdleExpiry extends BaseExpiry {
+  private static class TimeToIdleExpiry extends BaseExpiry<Object, Object> {
     TimeToIdleExpiry(Duration tti) {
       super(tti, tti, tti);
     }
   }
 
-  private static class NoExpiry extends BaseExpiry {
+  private static class NoExpiry extends BaseExpiry<Object, Object> {
 
     private static final Expiry<Object, Object> INSTANCE = new NoExpiry();
 
     private NoExpiry() {
       super(Duration.INFINITE, null, null);
     }
   }
+
+  /**
+   * Builder to create a simple {@link Expiry}.
+   *
+   * @param <K> Key type of the cache entries
+   * @param <V> Value type of the cache entries
+   */
+  public static final class ExpiryBuilder<K, V> {
+
+    private Duration create = Duration.INFINITE;
+    private Duration access = Duration.INFINITE;
```

Comment:
This is the main gripe I have with defaults. This would mean that unless set, all access makes the entry never expire. Seems really wrong as a default.
From the `Expiry` API, a sensible default is `null` here.
But that means all usage in tests needs to specify this value nearly always. Hence my doubts about making this utility public.